tests/test_misc.py

```diff
@@ -605,13 +605,13 @@ def test_multirpc(node_factory):
         b'{"id":5,"jsonrpc":"2.0","method":"listpeers","params":[]}',
         b'{"id":6,"jsonrpc":"2.0","method":"listpeers","params":[]}',
         b'{"method": "invoice", "params": [100, "foo", "foo"], "jsonrpc": "2.0", "id": 7 }',
-        b'{"method": "waitinvoice", "params": ["foo"], "jsonrpc" : "2.0", "id": 8 }',
```

Comment:
What?? This will return once `delinvoice` executes.  In fact, this is a key part of the test: we have different paths for commands that complete immediately, and commands that will defer, and we should test both paths. Both `invoice` and `delinvoice` exercise the path that completes immediately, while `waitinvoice` exercises the path where it defers.  It is important to check both.

Reply:
Ah, good catch!  I wonder why it didn't work, will debug!

Reply:
If I remember correctly, there was an issue once that a user complained that parallel JSON-RPC connections could run commands in parallel, but a single JSON-RPC connection could not, which is why we moved from processing commands linearly, to processing them in parallel. This test was intended to check precisely this change, I think. I will try to see if if I can dig up this issue.

Reply:
Here: https://github.com/ElementsProject/lightning/issues/934#issuecomment-364274318

The user was complaining about parallel `listpeers`, and not the pattern in that comment (which, it turns out, was mine, and is precisely what is tested in `test_multirpc`: `invoice` `waitinvoice` `delinvoice`).

However it seems a regression to me to remove our ability to run commands in parallel within a single connection.  If this feature is removed, then `multirpc` should also be removed outright, not "fixed" by not testing what it is intended to test in the first place.

Reply:
Hmm, I've thought some more about this, and I agree.  We should change the limit such that it only stop processing input if there's still stuff in the output buffer.

That's actually quite a large change, as it means resolving the case where two commands interleave their returns.  We need that change for notifications, but now it's important for commands as well.

I'll do this as a separate PR; for this one I'll simply note the FIXME.